src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -988,7 +989,13 @@ private static FormulaValue Substitute(EvalVisitor runner, EvalVisitorContext co
             var matchLen = match.Value.Length;
             var replacementLen = replacement.Value.Length;
 
-            var maxLenChars = SubstituteGetResultLength(sourceLen, matchLen, replacementLen, instanceNum < 0);
+            var maxLen = SubstituteGetResultLength(sourceLen, matchLen, replacementLen, instanceNum < 0);
+
+            if (!TryGetIntFromDouble(maxLen, out int maxLenChars))
+            {
+                throw new GovernorException($"Memory overuse: Used {maxLen} characters vs. {int.MaxValue} characters allowed.");
+            }

```

Comment:
we could catch overflow here and return a runtime error. 

CommonErrors.OverflowError